cou/steps/hypervisor.py

```diff
@@ -159,6 +160,16 @@ def get_azs(self, target: OpenStackRelease) -> AZs:
                 az = unit.machine.az or ""
                 azs[az].app_units[app.name].append(unit)
 
+            for s_unit in app.subordinate_units:
+                s_app = apps_dict[s_unit.name.split("/")[0]]
+                for machine in s_app.machines.values():
```

Comment:
Right now we don't have a map SubordinateUnit -> Machine as we have with a normal Unit. The idea was to not include units of an application that does not make part of the hypervisors machine. E.g: cinder/0 is deployed with nova-compute/0 but cinder/1 is deployed into a non-hypervisor machine. It's very unlikely this to happen, but we would skip that unit/machine that it's not a hypervisor.

Here this is not happening. If the subordinate unit has at least one unit in a hypervisor machine, all unis will be included.